waiter/config-full.edn

```diff
@@ -752,6 +752,9 @@
  ; ---------- Load Balancing ----------
 
  :work-stealing {
+                 ;; The maximum number of outstanding work-stealing offers per service per router:
+                 :max-work-stealing-offers-per-service 20
```

Comment:
It seems undesirable to not offer all available capacity to a lone router handling all the traffic, especially given that we have other ways of increasing the number of ports available to us.

Reply:
Does this setting belong in the service description, given the wide variety of `concurrency-level`s among services and therefore the wide variety of optimal values for this setting?

Reply:
We implemented work-stealing as an optimization. This configuration allows us to disable work-stealing (setting to `0`). We can also choose to make this value arbitrarily high if we have enough ports to play with.

Reply:
Agreed that it can be seen as useful setting as a value for 0 or infinity.  Unclear when it's set to anything in between.

Reply:
If we know we have 200 services running, and no more than 50 (for example) in-flight work-stealing offers per service, no more than 10K ports are in use the work-stealing mechanism. This is valuable in restricting the network resources used by the work-stealing mechanism.

Reply:
The problem is we don't know we have 200 services running.

Reply:
We can from other components. I can also change this to a global value where the limits are imposed on all work-stealing offers in flight.

Reply:
Yes we could.  We'd have to think about fair allocation of work steals across services, etc.  It's an easy step from here.

Realistically,
1. We're not going to turn work stealing off.  I'd rather have a :disabled flag for this anyway.
2. We can increase the number of ports available to us: https://making.pusher.com/ephemeral-port-exhaustion-and-how-to-avoid-it/  We need to do this anyway to create headroom for more services.
3. The work going on in #942 will hopefully dramatically reduce the number of work steals.

I don't see a need for this PR as it currently stands.  

Should we do the work now to make this a global limit?  It would be a nice to have, but we have many nice to haves that we probably want more.

Reply:
Maybe I'm missing some context on a specific instance where a service was sending out hundreds of work steals (that wasn't fixed by #942) and we simply don't want to provide that high of a level of service.  Let me know.

Reply:
If we can get an increase in the number of ports then this PR is unnecessary.
#942 will help, but not sure it alone will help unless we can increase port numbers.
As you said global limit is nice to have, but we can add it later.

Reply:
Created https://github.com/twosigma/waiter/pull/945 to enforce global limits on work-stealing offers.